geom/src/polyline.rs

```diff
@@ -624,23 +622,30 @@ impl PolyLine {
         assert_ne!(self, other);
 
         // There could be several collisions. Pick the "first" from self's perspective.
+        let mut closest_intersection: Option<(Pt2D, Angle)> = None;
+        let mut closest_intersection_distance: Option<Distance> = None;
+
         for l1 in self.lines() {
-            let mut hits: Vec<(Pt2D, Angle)> = Vec::new();
             for l2 in other.lines() {
                 if let Some(pt) = l1.intersection(&l2) {
-                    hits.push((pt, l1.angle()));
+                    if let Some(new_distance) = self.get_slice_ending_at(pt).map(|pl| pl.length()) {
+                        match closest_intersection_distance {
```

Comment:
about 1/2 the perf came from this method and 1/2 came from returning a line iterator instead of a Vec<Line>. 

If the changes in this method are too controversial for the 0.3 % speedup, then I'd still recommend keeping the `Iterator<Item = Line>` stuff which I think is less controversial.

Reply:
This seems simple enough!